feflow/utils/exceptions.py

```diff
@@ -0,0 +1,15 @@
+"""Module with custom exceptions that can be useful for including in protocols"""
+
+
+class ProtocolError(Exception):
+    """Base exception for OpenFE protocols"""
+
+
+class MethodConstraintError(ProtocolError):
+    """Custom exception to be raised when a fundamental limitation in the methodology is
+    that does not support."""
+
+
+class NotSupportedError(ProtocolError):
```

Comment:
This name is somewhat unspecific for something that inherits from something Protocol specific, could there be a better name?